modyn/config/schema/pipeline-schema.yaml

```diff
@@ -112,10 +112,22 @@ properties:
                 type: number
                 description: |
                   [AbstractDownsampleStrategy] percentage of points on which the metric (loss, gradient norm,..) is computed. Must be between 0 and 100
+              sample_before_batch:
+                type: boolean
+                description: |
+                  [AbstractDownsampleStrategy] If True, the samples are first sampled and then batched and supplied to the training loop. If False, the datapoints are first divided into batches and then sampled.
+              downsampling_period:
+                type: number
+                description: |
+                  [AbstractDownsampleStrategy] Points are downsampled every downsampling_period epochs (0, downsampling_period, 2*downsampling_period ...). Requires sample_before_batch = True and the default value is 1
               downsampled_batch_size:
                 type: number
                 description: |
-                  [AbstractDownsampleStrategy] How many points are sampled after computing the metric
+                  [AbstractDownsampleStrategy] How many points are sampled after computing the metric. Requires sample_before_batch set to False
+              downsampled_batch_ratio:
+                type: number
+                description: |
+                  [AbstractDownsampleStrategy] Ratio post_sampling_size/pre_sampling_size. For example, with 160 datapoints and a ratio of 50 we keep 80. Requires sample_before_batch set to True and must be in (0,100)
```

Comment:
yeah let's unify. so with a single parameter you mean a relative parameter? we can also have either relative or absolute, if you think that makes sense. but in any case it should not depend on BtS vs StB

Reply:
I think that the most flexible solution would be to have a relative parameter and an absolute hard cap. So something like size = min(dataset_size * relative, hard_cap)

Reply:
Yes, I agree. I will take a look at the code in a second

Reply:
err I will have to re check tomorrow. after going through the code, I am not sure I remember seeing this. So basically, we want an optional hard cap and if it is given, then size is implemented like to wrote.
```
size = min(dataset_size * relative, hard_cap) if hard_cap is not None else dataset_size * relative
```

Reply:
ok so actually this is the fist comment discussing this whole thing. I will leave it open since it mentions additional concepts such as hard cap and relative cap

Reply:
In fact, there has always been a hard cap, namely the parameter _limit_. And, when the presampling size is decided, the followning is done

```
def get_presampling_target_size(self) -> int:
  dataset_size = self._get_dataset_size()
  target_presampling = int(dataset_size * self.presampling_ratio / 100)
  
  if self.has_limit:
      target_size = min(self.training_set_size_limit, target_presampling)
  else:
      target_size = target_presampling
  
  return target_size
```

Reply:
Ahh, I think we were referring to two different things. With `limit`, one has a hard cap for the presampled dataset. I was referring to a hard cap of the downsampled dataset. But I think one can express that with `limit` as well, we know that the hard cap is downsampling ratio * limit for the downsampled set?

Reply:
Yes, exactly. There will never be more than downsampling ratio * limit samples.  If you prefer, I can add a downsampling_hard_cap parameter but I do not know if it will ever be used.
Presampling hard cap (aka `limit`) instead serves to avoid OOM problems in the downsampler